src/main/scala/viper/carbon/modules/MainModule.scala

```diff
@@ -31,6 +31,8 @@ trait MainModule extends Module {
     translateLocalVarSig(l.typ,l.localVar)
   }
 
+  def translateSMTFuncApp(fa: sil.SMTFuncApp): Exp
```

Comment:
I might be missing something, but I would put this in the FuncPredModule. Is there a particular reason you put it in the MainModule?